pythonsdk/src/main/python/vantiqservicesdk.py

```diff
@@ -53,54 +85,67 @@ def _status(self) -> dict:
-            for task in active_requests:
-                task.cancel()
+        with self._websocket_count.track_inprogress():
+            await websocket.accept()
+            active_requests: Set = set()
+
+            # Define a callback to remove the task from the active requests
+            def __complete_request(completed_task):
+                active_requests.discard(completed_task)
+                self._active_requests.dec(1)
+
+            try:
+                # Start by asking for our configuration (if we don't have it)
+                if self._client_config is None:
+                    config_request = {"requestId": CLIENT_CONFIG_MSG, "isControlRequest": True}
+                    await websocket.send_json(config_request, "binary")
+
+                while True:
+                    # Get the message in bytes and see if it is a ping
+                    msg_bytes = await websocket.receive_bytes()
+                    if msg_bytes == b'ping':
+                        await websocket.send_bytes('pong'.encode("utf-8"))
+                        continue
+
+                    # Spawn a task to process the message and send the response
+                    task = asyncio.create_task(self.__process_message(websocket, msg_bytes))
+
+                    # Add the task to the set of active requests and remove when done.
+                    # See https://docs.python.org/3/library/asyncio-task.html#creating-tasks
+                    active_requests.add(task)
+                    self._active_requests.inc(1)
+                    task.add_done_callback(__complete_request)
+
+            except WebSocketDisconnect:
+                pass
+
+            finally:
+                # Cancel all active requests
+                for task in active_requests:
+                    self._active_requests.dec(1)
+                    task.cancel()
 
     async def __process_message(self, websocket: WebSocket, msg_bytes: bytes) -> None:
         # Decode the message as JSON
         request = json.loads(msg_bytes.decode("utf-8"))
-        logging.debug('Request was: %s', request)
+        self._logger.debug('Request was: %s', request)
 
         # Set up default response and invoke the procedure
         response = {"requestId": request.get("requestId"), "isEOF": True}
+        procedure_name: str = 'unknown'
         try:
             # Get the procedure name and parameters
             procedure_name = request.get("procName")
             params = request.get("params")
 
             # Invoke the procedure and store the result
-            result = await self.__invoke(procedure_name, params)
-            response["result"] = result
+            with self._request_latency.labels(procedure_name).time():
+                result = await self.__invoke(procedure_name, params)
+                response["result"] = result
 
         except Exception as e:
+            self._logger.debug(f"Error invoking procedure {procedure_name}", exc_info=e)
```

Comment:
The `exc_info` arguments prints a stack trace which can be very handy in figuring out often obscure error messages.